### PR TITLE
Add  tesitng for modify and update

### PR DESCRIPTION
**Pull request by TayyabKM**
3 Tests were created. This tests for:

- editing the task description.
- updating an item's 'completed' status.
- the "Clear all completed" function.

### CODE REVIEW

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @TayyabKM, 

Good job so far 🔥!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 🔥🎉
- Add test file ✔
- No linter errors ✔

# Required Changes ♻️
- [ ] Installing dependencies is not working. You may check your package.json file and fix it
Screenshot 2022-11-25 at 11 27 33 AM

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/module/toDoDom.js [hunk 1]
```diff
@@ -1,58 +1,66 @@
...
+ import Data from './data.js';
+ import Remove from './remove.js';
+ import Modify from './modify.js';
+ import * as base from './base.js';
+ 
+ describe('Testing add and remove', () => {
+   test('Add function test', () => {
+     document.body.innerHTML = '<ul class="main-body-ul"></ul>';
+     const appList = document.querySelector('.main-body-ul');
+     toDoDom.data('task1', appList);
```
Comment (danielufeli) [inline src/module/toDoDom.js]: - [ ] Nice working on your tests, Kindly check it throws the error in the screenshot below when I run the test, 
image

**Review by danielufeli — CHANGES_REQUESTED** [review]
Hi @Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors ✔

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/module/toDoDom.js [hunk 2]
```diff
@@ -1,58 +1,66 @@
...
+     base.clearAllComplatedTasks();
+    
+     // Till this test we create 3 dom elements and change 1 element to complete one.
+     // That means when we use the clear all component button we will have 2 dom elements
+     // and 1 dom element that we created in the test for making checked true have been deleted
+    
+     const completedTasks = Data.dataGet().length;
+     expect(completedTasks).toBe(2);
+   });
+ });
```
Comment (tugsem) [inline src/module/toDoDom.js:66]: - [ ] As requested by the previous reviewer make sure all of your tests are passing: 
image

**Review by tugsem — COMMENTED** [review]
## STATUS: REQUIRED CHANGES ♻️
Hi @TayyabKM and @Itswali , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**